### ISSUE TITLE
'Amount to Borrow' Field, 'Maximum' Borrow Field / 'Max' Button Behavior Not Functioning Correctly

### ISSUE THREAD

**Issue by notwheel** [script3/yieldblox-beta#173]
tl;dr: The Amount to Borrow field doesn't update correctly or as expected when making changes to the other parameters and  the Maximum Borrow amount essentially breaks after Deposit Additional Collateral is used.
 

When you clicking Max we're expecting the Maximum amount you can borrow is filled in, like so:

- <img width="653" alt="Screen Shot 2021-10-08 at 02 05 28" src="https://user-images.githubusercontent.com/543510/136506367-23647d58-395b-42e7-a0ab-5e6d181933f4.png">

If you toggle on Deposit Additional Collateral, enter a value for a additional funds, and click Max again, the value updates correctly:

- <img width="653" alt="Screen Shot 2021-10-08 at 02 06 20" src="https://user-images.githubusercontent.com/543510/136506468-83e4f994-ce17-4a60-b0cd-76eaa935fc5c.png">

If you then toggle off Deposit Additional Collateral, the old calculate Max value remains in the Amount to Borrow and in Maximum Borrow (it _should_ re-calculate and fill in the actual amount you can borrow):

- <img width="653" alt="Screen Shot 2021-10-08 at 02 08 40" src="https://user-images.githubusercontent.com/543510/136506721-36cb8c47-e195-4a16-81c0-6a031c4c6afe.png">

Now, after you exit or click off of that dialog and open any other asset, we're still populating the old Amount to Borrow. For example, with something like BTC, you then have a huuuuge difference in asset price (note that the Maximum Borrow is correct in this example, but Amount to Borrow is still the amount from the old ETH dialog:

- <img width="653" alt="Screen Shot 2021-10-08 at 02 09 30" src="https://user-images.githubusercontent.com/543510/136506786-f269fd2d-f61e-4e8f-b77b-a486a5a93e44.png">

It can be confusing and if one isn't paying attention they can quite easily attempt to borrow an absurd amount of an asset. 

The ideal behavior would seem to be clearing the field when the dialog for that asset closes AND when the Deposit Additional Collateral toggle changes state, as well as updating the Maximum Borrow amount.

Tried in incognito as well.
Refreshing or changing to Lend view seems to clear it.

**Comment by mootz12** [comment]
Thanks for the detailed bug! We have resolved this with our updated repay modal.

**Comment by notwheel** [comment]
> Thanks for the detailed bug! We have resolved this with our updated repay modal.

You are very welcome!
I have a question - How do we connect our bug reports to our Discord accounts?

**Comment by notwheel** [comment]
**Discord username: _notwill#7468_**

Per the latest announcement regarding bugs - this is my username. Please let me know if you need anything else!

**Comment by markuspluna** [comment]
Bug Bounty paid in transaction:

https://laboratory.stellar.org/#explorer?resource=transactions&endpoint=single&values=eyJ0cmFuc2FjdGlvbiI6IjJkNzVjZTM1YTdiYTdhOTViYzczMjk1MTE2M2RlODQ0MWQ0NzY5Y2Q3OGY3NTY1ODg3MmFmZWZkZjdhNTVlMzAifQ%3D%3D&network=public
